src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenDeconstructTests.cs

```diff
@@ -1355,6 +1435,71 @@ public static void Deconstruct(this C value, out int a, out string b)
         }
 
         [Fact]
+        public void UnderspecifiedDeconstructGenericExtensionMethod()
+        {
+            string source = @"
+class C
+{
+    static void Main()
+    {
+        long x;
+        string y;
+        (x, y) = new C();
+    }
+}
+static class Extension
+{
+    public static void Deconstruct<T>(this C value, out int a, out T b)
+    {
+        a = 2;
+        b = default(T);
+    }
+}";
+
+            var comp = CreateCompilationWithMscorlib(source, references: new[] { ValueTupleRef, SystemRuntimeFacadeRef, SystemCoreRef });
+            comp.VerifyDiagnostics(
+                // (8,18): error CS0411: The type arguments for method 'Extension.Deconstruct<T>(C, out int, out T)' cannot be inferred from the usage. Try specifying the type arguments explicitly.
+                //         (x, y) = new C();
+                Diagnostic(ErrorCode.ERR_CantInferMethTypeArgs, "new C()").WithArguments("Extension.Deconstruct<T>(C, out int, out T)").WithLocation(8, 18),
```

Comment:
We look-up Deconstruct as-if you invoked `Deconstruct(out var x1, out var x2, ...)`. Any types we may know from the left-hand-side don't contribute.
We discussed it in LDM earlier this week (which is why I wanted to record the tests) and decided that is ok for now.